tests/test_site_apportionment/test_site_apportionment.py

```diff
@@ -634,3 +636,127 @@ def test_sort_rows_missing_cols(self, caplog):
 
         with pytest.raises(KeyError):
             sort_rows_order_cols(input_df, cols_in_order)
+
+
+class TestWeightValues(object):
+    """Tests for weight_values."""
+    
+    @pytest.fixture(scope="function")
+    def weight_values_test_df(self):
+        """Test data for weight_values."""
+        frame = {
+            "reference": [0, 1, 2, 3, 4],
+            "val_col_1": [10, 14, 16, 18, 6],# test using ints rather than float
+            "val_col_2": [3.5, 4.5, 10, np.nan, 0],# test np.nan+0+floats
+            "weight_col": [1.5, 1.5, 1.0, 2.0, 3.0]
+                }
+        return pandasDF(frame)
+    
+    @pytest.fixture(autouse=True)
+    def set_attrs(self):
+        """Set class attributes to pass to function."""
+        self.value_cols = ["val_col_1", "val_col_2"]
+        self.weight_col = "weight_col"
+
+    def test_weight_values_raises(self, weight_values_test_df):
+        """Tests for raises in weight_values."""
+        with pytest.raises(KeyError, match=".*not_weight_col.*"):
+            weight_values(weight_values_test_df, 
+                          self.value_cols, 
+                          "not_weight_col")
+
+    def test_weight_values_on_pass(self, weight_values_test_df):
+        """General tests for weight_values."""
+        output = weight_values(weight_values_test_df,
+                                self.value_cols, 
+                                self.weight_col)
+        assert np.array_equal(output["val_col_1"],
+                       [15.0, 21.0, 16.0, 36.0, 18.0])
+        assert np.array_equal(output["val_col_2"],
+                       [5.25, 6.75, 10.0, np.nan, 0.0], equal_nan=True)
+        
```

Comment:
Again, this is neat but I'd prefer our  style of input df and exp_output_df. This will still adequately test and give an at-a-glance illustration of the function

Reply:
Hi @CBROWN-ONS , would you please update this to our house style and write columns followed by data in the same way as you've done for create_category_df test below? Thanks

Reply:
pg_numeric and instance to int please